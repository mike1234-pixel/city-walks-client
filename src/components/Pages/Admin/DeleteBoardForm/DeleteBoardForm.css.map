{"version":3,"sourceRoot":"","sources":["DeleteBoardForm.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA","file":"DeleteBoardForm.css"}