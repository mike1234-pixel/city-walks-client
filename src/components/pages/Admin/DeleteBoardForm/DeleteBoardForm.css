/** @format */

.delete-board-form {
  padding: 20px;
}
