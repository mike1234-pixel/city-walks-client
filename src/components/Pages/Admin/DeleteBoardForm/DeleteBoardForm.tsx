import React, { useState } from "react";
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact";
import axios, { AxiosError, AxiosResponse } from "axios";
import toTitleCase from "../../../../functions/toTitleCase";
import SelectedBoard from "../../../../types/PostRequests/SelectedBoard";
import "./DeleteBoardForm.css";

const DeleteBoardForm = () => {
  const [boardName, setBoardName] = useState<string>("");

  const handleChange: (event: React.ChangeEvent<any>) => void = (event) => {
    setBoardName(event.target.value);
  };

  const handleSubmit: (event: React.FormEvent) => void = (event) => {
    event.preventDefault();

    const payload: SelectedBoard = {
      boardName: toTitleCase(boardName),
    };

    axios
      .delete("https://city-walks.herokuapp.com/delete-board", {
        data: payload,
      })
      .then((res: AxiosResponse) => {
        if (res.data === "board deleted") {
          console.log("board deleted");
        } else {
          console.log("board not deleted");
        }
      })
      .catch((err: AxiosError) => {
        console.log(err);
      });

    alert("Board Deleted");
    setBoardName("");
    window.scrollTo(0, 0);
  };

  return (
    <div>
      <h2>Delete Board</h2>
      <form onSubmit={handleSubmit} className='delete-board-form'>
        <MDBInput
          type='text'
          name='board-name'
          id='board-name'
          value={boardName}
          label='board name'
          onChange={handleChange}
          required
        />
        <MDBBtn outline color='elegant' type='submit'>
          Delete Board <MDBIcon icon='trash' />
        </MDBBtn>
      </form>
    </div>
  );
};

export default DeleteBoardForm;
