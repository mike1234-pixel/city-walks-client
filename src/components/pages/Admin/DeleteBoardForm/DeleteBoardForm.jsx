import { useState } from "react"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import axios from "axios";
import toTitleCase from '../../../../functions/toTitleCase'
import './DeleteBoardForm.css'

const DeleteBoardForm = () => {

    const [boardName, setBoardName] = useState("")

    const handleChange = (event) => {
        setBoardName(event.target.value)
    }

    const handleSubmit = (event) => {
        event.preventDefault()

        const payload = {
            boardName: toTitleCase(boardName)
        }

        axios
        .delete("http://localhost:5000/delete-board", {data: payload})
        .then((res, err) => {
            if (err) {
              console.log(err);
            } else if (res.data === "board deleted") {
              console.log("board deleted")
            } else {
              console.log("board not deleted")
            }
          });
  
          alert("Board Deleted")
          setBoardName("")
          window.scrollTo(0, 0);
    } 


    return (
        <div>
        <h2>Delete Board</h2>
        <form onSubmit={handleSubmit} className="delete-board-form">
            <MDBInput type="text" name="board-name" id="board-name" value={boardName} label="board name" onChange={handleChange} required/>
            <MDBBtn outline color="elegant" type="submit">
                Delete Board <MDBIcon icon="trash" />
          </MDBBtn>
        </form>
    </div>
    )
}

export default DeleteBoardForm