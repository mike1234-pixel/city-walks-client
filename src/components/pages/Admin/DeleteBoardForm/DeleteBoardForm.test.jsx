import renderer from 'react-test-renderer'
import DeleteBoardForm from "./DeleteBoardForm"

// snapshot test
it('DeleteBoardForm Page renders correctly', () => {  

  const tree = renderer
    .create(
        <DeleteBoardForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});