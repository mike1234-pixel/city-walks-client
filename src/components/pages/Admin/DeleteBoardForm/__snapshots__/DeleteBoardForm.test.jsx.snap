// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteBoardForm Page renders correctly 1`] = `
<div>
  <h2>
    Delete Board
  </h2>
  <form
    className="delete-board-form"
    onSubmit={[Function]}
  >
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="board-name"
        name="board-name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="board-name"
        id=""
        onClick={[Function]}
      >
        board name
      </label>
    </div>
    <button
      className="btn-elegant btn-outline-elegant btn Ripple-parent"
      data-test="button"
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Delete Board 
      <i
        className="fa fa-trash"
        data-test="fa"
      />
      <div
        className="Ripple Ripple-outline "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </form>
</div>
`;
